manager/recurringjob.go

```diff
@@ -114,9 +114,64 @@ func (m *VolumeManager) UpdateRecurringJob(spec longhorn.RecurringJobSpec) (*lon
 }
 
 func (m *VolumeManager) DeleteRecurringJob(name string) error {
+	if err := m.CleanVolumeRecurringJob(name); err != nil {
+		return err
+	}
+
 	if err := m.ds.DeleteRecurringJob(name); err != nil {
 		return err
 	}
 	logrus.Infof("Deleted recurring job %v", name)
 	return nil
 }
+
+func (m *VolumeManager) CleanVolumeRecurringJob(name string) error {
+	recurringJob, err := m.ds.GetRecurringJob(name)
+	if err != nil {
+		return err
+	}
+
+	// check if other recurring jobs contains same groups
+	recurringJobs, err := m.ds.ListRecurringJobs()
+	if err != nil {
+		return err
+	}
+	rmGroups := []string{}
+	for _, group := range recurringJob.Spec.Groups {
+		inUse := 0
+		for _, obj := range recurringJobs {
+			if util.Contains(obj.Spec.Groups, group) {
+				inUse++
+			}
+			if inUse > 1 {
+				continue
+			}
+		}
+		if inUse == 1 {
+			rmGroups = append(rmGroups, group)
+		}
+	}
+
+	// delete volume labels
+	volumes, err := m.ds.ListVolumes()
+	if err != nil {
+		return err
+	}
+	for volName, vol := range volumes {
+		list, err := m.ListVolumeRecurringJob(volName)
+		if err != nil {
+			return err
+		}
+		if vol.Spec.LastAttachedBy == name && vol.Status.State != longhorn.VolumeStateDetached {
+			m.Detach(volName, "")
+		}
+		for jobName := range list {
```

Comment:
Is it job name duplicated? like the same name for a job or a group? maybe you can leverage job properties to do the condition instead of job name? WDYT?

Reply:
Good catch, I missed this one. Fixed and also added to the automation test case in this [PR](https://github.com/longhorn/longhorn-tests/pull/891).